### PR TITLE
Document changes related to continuous localization

### PR DESCRIPTION
**Pull request by nathandyer**
## Status

Work in progress
- [x] What `i18n` repositories and branches does this workflow obsolete?
    - freedomofpress/infrastructure#4479
- [x] https://github.com/freedomofpress/securedrop/pull/7004#pullrequestreview-1700839126

Closes https://github.com/freedomofpress/securedrop/issues/6916

## Description of Changes

* Description: Documents new changes to the workflow as a result of the switch to continuous localization

* Related Issues

[SecureDrop #6232](https://github.com/freedomofpress/securedrop/issues/6232)

## Testing
TK

## Release 

After merge:
* [ ] Replace <https://github.com/freedomofpress/securedrop-client/wiki/Internationalization> with a link to `translations.rst`.

## Checklist (Optional)

- [x] Doc linting (`make docs-lint`) passed locally
- [x] Doc link linting (`make docs-linkcheck`) passed
- [x] You have previewed (`make docs`) docs at http://localhost:8000

### CODE REVIEW

File: docs/i18n.rst
Comment (cfm) [inline docs/i18n.rst:69]: Some things to clarify and/or expand:

```suggestion
appear in Weblate, and translation can begin. Translation and review can take
place continuously, at any time. Translations for supported languages [link] are
reviewed and finalized during the release process.
```

As in <https://github.com/freedomofpress/securedrop-client/wiki/Internationalization>, I think our goal should be that translations get code-merged more rather than less often.  This will mean (a) [some translations get code-merged before they've been translation-reviewed](https://github.com/freedomofpress/securedrop-client/pull/1548#discussion_r1156375077) and (b) some translations will be merged for unsupported languages, before language support is evaluated as part of the release process.

File: docs/i18n.rst
Comment (cfm) [inline docs/i18n.rst:104]: Per below, we'll name this target more explicitly going forward (and elsewhere in this document):

```suggestion
files should also be updated by running ``make extract-strings`` in the root
```

Yet to be determined: whether this is enforced at commit-time or only in CI.

File: docs/i18n.rst
Comment (cfm) [inline docs/i18n.rst:109]:
```suggestion
The ``extract-strings`` target gathers source strings, then updates the 
`.pot`_ files for the SecureDrop server code and the desktop icons.
```

The new approach defers the `.pot`-to-`.po` merge to Weblate.  See freedomofpress/securedrop-client#1657 for a tidy example.  This would also remove the paragraph beginning "Then review...".

File: docs/i18n.rst
Comment (cfm) [inline docs/i18n.rst:132]: This paragraph can be removed: source strings will be updated in the same branch (if not the very same commits) in which the enclosing source code is changed.

File: docs/i18n.rst
Comment (cfm) [inline docs/i18n.rst:136]: A number of steps here can disappear entirely:
* Merge develop into the Weblate fork (and references in "Two weeks before the release: string freeze")
* Merge translations back to develop (and references in "Release day")
* Merge translations back to develop

File: docs/i18n.rst
Comment (cfm) [inline docs/i18n.rst:251]: This section describes the `.po`→`.mo` compilation rather than the `.py`→`.po[t]` extraction.  But it can go away too: if we're not letting Weblate compile our `.mo` files, it'll be part of `make dev` and/or `dev-deps` (TBD).

**Review by cfm — CHANGES_REQUESTED** [review]
@nathandyer, here's a first review of your first draft!  I think this DDD ("documentation-driven development") approach is working out nicely.  There are a few things we can clarify and expand here and there, but it turns out there's a lot _more_ we can delete altogether.  :-)

The one higher-level question I'd like to pose to you is whether we should incorporate some version of the overview in <https://github.com/freedomofpress/securedrop-client/wiki/Internationalization>.  I'm keen to have a unified _architecture_ for localization across the Client and Server, as well as common workflows and steps for executing it, and I think documenting that architecture centrally will be key to holding to it.  What do you think?

File: docs/i18n.rst
Comment (nathandyer) [reply docs/i18n.rst:109]: That makes sense, thanks for clarifying!

**Comment by nathandyer** [comment]
Thanks for the initial review @cfm! I agree, so far I'm a big fan of the documentation-driven development approach! I'll be sure to go through and address all your line-level comments, but I wanted to make sure to answer your larger question before I sign off for the day, so we can plan our work accordingly.

I agree that incorporating an overview [like this](https://github.com/freedomofpress/securedrop-client/wiki/Internationalization) would be ideal, so that these newly unified processes can be documented in one central location. I'm going to set aside a period of time tomorrow morning to work on that, and should have it ready for you to take a look when you begin your day.

File: docs/i18n.rst
Comment (cfm) [inline docs/i18n.rst:1]: This is looking good.  The section "Push your branch and create a pull request" can also go!

**Review by cfm — COMMENTED** [review]
Thanks, @nathandyer!  This draft looks great to me.  I think the easiest thing will be to keep this marked as draft until freedomofpress/securedrop#6917 is done, and then (modulo any nits or tweaks along the way) we can ask the reviewer (or a second, if it's you) to look at both of them together.  That seems like the simplest way to prevent the implementation and this documentation from getting out of sync.

**Comment by nathandyer** [comment]
@cfm  With the 2.7.0 release officially out in the world, are there any areas where the process diverged from what we have documented here? Is there anything else we need to change before flipping this to ready for review?

**Comment by cfm** [comment]
Back to you, @nathandyer!

File: docs/i18n.rst
Comment (nathandyer) [inline docs/i18n.rst:232]: Should `source-string` be hyphenated here?

**Comment by nathandyer** [comment]
Thank you so much @cfm, this all looks great. I left one small nitpick (a question really), but otherwise it LGTM.

This still shows as "Changes requested" from your review, but as far as I'm concerned, this is ready for an approve/merge, pending that one possible change mentioned above. If you feel this is ready, please don't hesitate to push the big scary button :)

File: docs/i18n.rst
Comment (cfm) [reply docs/i18n.rst:232]: I believe so: "source-string" modifies "suggestions", rather than "source" modifying "string suggestions".

**Review by cfm — APPROVED** [review]

**Comment by cfm** [comment]
Thanks, @nathandyer!  Merging now, knowing that we can always revise later...or as soon as #112.  :-)

**Pull request merged**
